src/main/scala/pl/datart/csvtojson/service/Metrics.scala

```diff
@@ -0,0 +1,5 @@
+package pl.datart.csvtojson.service
+
+import nl.grons.metrics4.scala.DefaultInstrumented
+
+object Metrics extends DefaultInstrumented
```

Comment:
Isn't it a global mutable state? What's the point of dealing with `tasks: Ref[F, Map[TaskId, Task]` when we could define it as:

```
object TaskStorage {
   val tasks = TrieMap[TaskId, Task]
}
```   
?

Reply:
Well, when using either `DefaultInstrumented` or custom registries its standard usage looks (more or less) like that. I'm referring to:
https://github.com/erikvanoosten/metrics-scala/blob/master/docs/Manual.md#custom-registries

So it introduces state. The line count could be encapsulated in `Task` class and accessed over `tasks` `Ref`, but having it incremented for each line I found it more like _metrics-like_ use case.